src/base/memory_monitor_sr.cpp

```diff
@@ -402,29 +452,82 @@ namespace cubperf
     m_mmon = mmon;
   }
 
-  int memory_monitor::aggregater::get_server_info (const MMON_SERVER_INFO &info)
+  void memory_monitor::aggregater::get_server_info (MMON_SERVER_INFO &info) const
   {
-    return 0;
+    strncpy (info.name, m_mmon->m_server_name.c_str (), m_mmon->m_server_name.size ());
+    info.total_mem_usage = m_mmon->m_total_mem_usage.load ();
   }
 
-  int memory_monitor::aggregater::get_module_info (const MMON_MODULE_INFO &info, int module_index)
+  void memory_monitor::aggregater::get_module_info (MMON_MODULE_INFO *&info, int module_index) const
   {
-    return 0;
+    if (module_index == -1)
+      {
+	// aggregate all detail memory information of modules
+	for (int idx = 0; idx < MMON_MODULE_LAST; idx++)
+	  {
+	    m_mmon->m_module[idx]->aggregate_stats (info[idx], false);
+	  }
+      }
+    else
+      {
+	m_mmon->m_module[module_index]->aggregate_stats (info[0], false);
+      }
   }
 
-  int memory_monitor::aggregater::get_transaction_info (const MMON_TRAN_INFO &info, int tran_count)
+  void memory_monitor::aggregater::get_module_info_summary (MMON_MODULE_INFO *&info, bool sorted_by_mem_usage) const
   {
-    return 0;
+    for (int i = 0; i < MMON_MODULE_LAST; i++)
+      {
+	m_mmon->m_module[i]->aggregate_stats (info[i], true);
+      }
+
+    if (sorted_by_mem_usage)
+      {
+	const auto &comp = [] (const auto& module_info1, const auto& module_info2)
+	{
+	  return module_info1.stat.cur_stat > module_info2.stat.cur_stat;
+	};
+	std::sort (info, info + MMON_MODULE_LAST, comp);
+      }
   }
 
-  void memory_monitor::add_stat (MMON_STAT_ID stat_id, int64_t size)
+  void memory_monitor::aggregater::get_transaction_info (MMON_TRAN_INFO &info, int tran_count) const
   {
-    int module_idx = get_module_idx (stat_id);
+    std::vector <std::pair <int, uint64_t>> tran_info;
+
+    logtb_get_tran_memory_info_nolatch (tran_info);
+
+    if (tran_count == 0)
+      {
+	// transaction option with no tran-count option
+	info.num_tran = tran_info.size ();
```

Comment:
> I added comments with branching based on the options because the explanation could be confusing. Default option of memmon has to show 5 transaction in descending order (that means, tran_count = 5).

1. When tran_count is not given, I was told that only 5 trxs are shown, wasn't it? I barely remember, but the comment above said that.. so I'm asking.
2. Aren't it needed to be sorted?